client/environment_test.go

```diff
@@ -331,3 +386,29 @@ var _ = Describe("Environment Client", func() {
 		})
 	})
 })
+
+func TestMarshalEnvironmentCreateWithoutTemplate(t *testing.T) {
+	templateCreate := TemplateCreatePayload{
+		Name: "name",
+		SshKeys: []TemplateSshKey{
+			{Id: "id1", Name: "name1"},
+		},
+		Type: "terraform",
+	}
+	environmentCreate := EnvironmentCreate{
+		Name:      "name",
+		ProjectId: "project_id",
+	}
+
+	environmentCreateWithoutTemplate := EnvironmentCreateWithoutTemplate{
+		EnvironmentCreate: environmentCreate,
+		TemplateCreate:    templateCreate,
+	}
+
+	b, err := json.Marshal(environmentCreateWithoutTemplate)
+	require.NoError(t, err)
+	s := string(b)
+	require.True(t, strings.Contains(s, `"sshKeys":[{"id":"id1","name":"name1"}]`))
+	require.True(t, strings.Contains(s, `"name":"name"`))
```

Comment:
this test looks odd, as it doesn't really assert that the flattening works (as those strings should also be present for the nested form of it)

Reply:
@ItamarMalka 

ok. Let me improve it.
I'll send an update shortly.

Reply:
@ItamarMalka - good catch it actually found a bug (after changing the test).
This is now fixed. 

d2f17f686e17d0164db78efb1bf7668e7bf75ee8

Reply:
what about [this](https://github.com/env0/terraform-provider-env0/pull/469#discussion_r943600924) for unmarshaling this won't work?

Reply:
@ItamarMalka - sorry I missed your question. Responded in the other thread.